cluster-autoscaler/simulator/scheduler_based_predicates_checker_test.go

```diff
@@ -0,0 +1,152 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package simulator
+
+import (
+	"testing"
+	"time"
+
+	. "k8s.io/autoscaler/cluster-autoscaler/utils/test"
+
+	"github.com/stretchr/testify/assert"
+
+	apiv1 "k8s.io/api/core/v1"
+)
+
+func TestCheckPredicate(t *testing.T) {
+	p450 := BuildTestPod("p450", 450, 500000)
+	p600 := BuildTestPod("p600", 600, 500000)
+	p8000 := BuildTestPod("p8000", 8000, 0)
+	p500 := BuildTestPod("p500", 500, 500000)
+
+	n1000 := BuildTestNode("n1000", 1000, 2000000)
+	SetNodeReadyState(n1000, true, time.Time{})
+
+	tests := []struct {
+		name          string
+		node          *apiv1.Node
+		scheduledPods []*apiv1.Pod
+		testPod       *apiv1.Pod
+		expectError   bool
+	}{
+		{
+			name:          "other pod - insuficient cpu",
+			node:          n1000,
+			scheduledPods: []*apiv1.Pod{p450},
+			testPod:       p600,
+			expectError:   true,
+		},
+		{
+			name:          "other pod - ok",
+			node:          n1000,
+			scheduledPods: []*apiv1.Pod{p450},
+			testPod:       p500,
+			expectError:   false,
+		},
+		{
+			name:          "empty - insuficient cpu",
+			node:          n1000,
+			scheduledPods: []*apiv1.Pod{},
+			testPod:       p8000,
+			expectError:   true,
+		},
+		{
+			name:          "empty - ok",
+			node:          n1000,
+			scheduledPods: []*apiv1.Pod{},
+			testPod:       p600,
+			expectError:   false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var err error
+			predicateChecker, err := NewTestPredicateChecker()
+			clusterSnapshot := NewBasicClusterSnapshot()
+			err = clusterSnapshot.AddNodeWithPods(tt.node, tt.scheduledPods)
+			assert.NoError(t, err)
+
+			predicateError := predicateChecker.CheckPredicates(clusterSnapshot, tt.testPod, FakeNodeInfoForNodeName(tt.node.Name))
+			if tt.expectError {
+				assert.NotNil(t, predicateError)
+				assert.Equal(t, NotSchedulablePredicateError, predicateError.ErrorType())
+				assert.Equal(t, "Insufficient cpu", predicateError.Message())
+				assert.Contains(t, predicateError.VerboseMessage(), "Insufficient cpu; predicateName=NodeResourcesFit")
+			} else {
+				assert.Nil(t, predicateError)
+			}
+		})
+	}
+}
+
+func TestFitsAnyNode(t *testing.T) {
+	p900 := BuildTestPod("p900", 900, 1000)
+	p1900 := BuildTestPod("p1900", 1900, 1000)
+	p2100 := BuildTestPod("p2100", 2100, 1000)
+
+	n1000 := BuildTestNode("n1000", 1000, 2000000)
+	n2000 := BuildTestNode("n2000", 2000, 2000000)
+
+	clusterSnapshot := NewBasicClusterSnapshot()
+	_ = clusterSnapshot.AddNode(n1000)
+	_ = clusterSnapshot.AddNode(n2000)
```

Comment:
Any reason for assigning to _, rather than just calling the function with no assignment (or asserting NoError on the result of AddNode)?

Reply:
Just an intellij warning that we are ignoring return value.
I will actually test assert NoError here.